test/cpp/end2end/end2end_test.cc

```diff
@@ -875,6 +875,25 @@ TEST_P(End2endTest, RequestStreamTwoRequests) {
   EXPECT_TRUE(s.ok());
 }
 
+// This was added to prevent regression from issue:
+// https://github.com/grpc/grpc/issues/12975
+TEST_P(End2endTest, RequestStreamTwoRequestsWithFirstWriteBufferHint) {
+  MAYBE_SKIP_TEST;
+  ResetStub();
+  EchoRequest request;
+  EchoResponse response;
+  ClientContext context;
+
+  auto stream = stub_->RequestStream(&context, &response);
+  request.set_message("hello");
+  EXPECT_TRUE(stream->Write(request, WriteOptions().set_buffer_hint()));
```

Comment:
So in with this PR, are we just silently changing options.set_buffer_hint() to options.clear_buffer_hint() if it's not the last message? Can we at least add some logs?